internal/provider/provider.go

```diff
@@ -147,31 +142,68 @@ func (p *PrefectProvider) Configure(ctx context.Context, req provider.ConfigureR
 			fmt.Sprintf("The Prefect API Endpoint %q is not a valid URL: %s", endpoint, err),
 		)
 	}
+	isPrefectCloudEndpoint := endpointURL.Host == "api.prefect.cloud" || endpointURL.Host == "api.prefect.dev" || endpointURL.Host == "api.stg.prefect.dev"
 
+	// Extract the API Key from configuration or environment variable.
 	var apiKey string
-	if !config.APIKey.IsUnknown() {
```

Comment:
we already perform unknown checks on these config values in the lines above

in this case, we want to do a null check.  the unknown values would come into play if we're passing in provider configurations that are dependent on other resource attributes that aren't yet know (like the id of a resource that will be created in the same plan)